### PR TITLE
Failed instances table bug fix

### PR DESCRIPTION
**Pull request by dansterenson**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43324

## Description
Until now when no "Failed Instances" were found, the result list contained one element of empty dict causing an empty row in the Failed Instances table. The empty dict was entered in the first place because "SetGridField.py" did not handle empty values properly. Now "SetGridField.py" handles empty values.

## Screenshots
image

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by reutshal — APPROVED** [review]
Nice 🚀

**Comment by dansterenson** [comment]
@amshamah419 Can you please go over the RN?

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 1]
```diff
@@ -167,7 +167,8 @@ def validate_entry_context(context_path: str, entry_context: Any, keys: List[str
                 'of type dict.')
         else:
             return
-
+    if entry_context is None:
+        entry_context = []
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:171]:
```suggestion

    if entry_context is None:
        entry_context = []

```

**Review by yaakovi — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@yaakovi @reutshal @David-BMS Doc review completed.

**Comment by David-BMS** [comment]
@altmannyarden Please validate it works in the health check pack

File: Packs/CommonScripts/Scripts/script-FailedInstances.yml [hunk 2]
```diff
@@ -61,7 +61,8 @@ script: >
   var total = (countSuccess + countFailed).toString();
 
 
-  if (failedInstances.length === 0) {
+  if (countFailed === 0) {
+      failedInstances.push({})
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/script-FailedInstances.yml:65]: Please add a comment to explain the logic

Comment (dansterenson) [reply Packs/CommonScripts/Scripts/script-FailedInstances.yml:66]: Done

**Review by reutshal — APPROVED** [review]

**Review by reutshal — APPROVED** [review]

**Pull request merged**